### DISCUSSION TITLE
Outliner layout X Mindmap layout

### DISCUSSION THREAD

**Discussion by euu2021** [freeplane/freeplane#682] [General]
Using the new layout that places the parent on top, I realized what's the difference between the outliner layout and the mindmap layout that makes me feel more comfortable using the later.

The outliner layout is, by far, the prevailing one, for example, in text editors and word processors (and, of course, in self-proclaimed outliners). Now, the question is that the mindmap layout is more efficient in showing the parent and the children at the same time.

For example, at the same zoom level, in the outliner mode, the visual contact with the parent is lost after node 16 (the screenshot is in the limit of the boundaries of the screen). On the other hand, on the mindmap layout, there is visual contact from nodes 1 to 31 :

![image](https://user-images.githubusercontent.com/77707706/192071087-1ce9c93b-edc1-4650-b5bc-5730376107a5.png)

This is relevant, because in this kind of information organization (Parent->Child), the back and forth visual and mental swing between the child and the parent is essential, in order to get the sense of context, and also relieve the short term memory (because, then, the user don't need to keep in the short term memory which is the parent of the current node that is being read).

So, I just would like to share this insight that I had and made me understand why I always felt more comfortable when using Freeplane in comparison with outliners.

**Comment by awolo** [comment]
Interesting. I like your wanting to have the parent in view as long/a easily as possible. I wonder if there's a feature request here, like "Context View", a version of the Map Overview that, instead of always showing everything, shows what would be the inverse of "Jump In": always showing the hierarchical ancestor map of the currently selected node? Without needing to create some kind of Filter. And the technology is already written to make that reminder area as small or large as you want. So, when I click on a Node, Context Overview would just show from root down into that node, though just the lineage of that node and not any cousins. Hmm.

Another point, about my comfort level using FP:
When I'm looking at a group of siblings that doesn't fit on the display, it ususally means I don't want to really look at each one individually. If I do and just haven't gotten to it, I try to remember that, "Oh, it's time to see if I should make some groups here". Sometimes that will make it easier to deal with the teeming children.

Also, this is just why I enjoy using FP and keep coming back to it no matter what other tool I investigate: you can so easily switch between views, create new subgroups, select some to change their Style and let them remain in the current list and sequence yet still be grouped visually because they share a Style, ....

Do you often have groups that size in your example that you want to keep as one long list? I'd love to see an actual example.

On the other hand, I love lists. My mind often will pour out a stream of more or less disjoint thoughts and I want to just get them down. In that case, actually and again just for me, I want to start at the top and go through them and refactor them some way, not keep them as a long list.

It reminds me of the saying, "Not Either Or, but Both And". What we have in FP is definitely Both And! if you know what I mean. We can have all kinds of different ways into the data, and our Leader DP always strives to make it friendly and intuitive to use lots of them, easily..

Thanks for this thread, I love thinking and talking about this stuff that usually goes on pretty unconsciously as I use FP.

**Reply by euu2021** [reply]
> I wonder if there's a feature request here, like "Context View", a version of the Map Overview that, instead of always showing everything, shows what would be the inverse of "Jump In": always showing the hierarchical ancestor map of the currently selected node? Without needing to create some kind of Filter. And the technology is already written to make that reminder area as small or large as you want. So, when I click on a Node, Context Overview would just show from root down into that node, though just the lineage of that node and not any cousins. Hmm.


I think a good example of this is the predecessors panel on yEd:
![image](https://user-images.githubusercontent.com/77707706/192171045-ff6b4ead-cae6-49f1-b50c-b17fd703205f.png)

I never used yEd for outlining, so I don't know how is the experience of using this panel, but it may be something to consider. 

> Do you often have groups that size in your example that you want to keep as one long list? I'd love to see an actual example.

For me, the size of the group of children gets a problem because of the size of each node, and not so much because of the number of nodes. When doing research, for example, the user needs to deal with large nodes, consisting of paragraphs and quotes. See, for example,[ this post of mine](https://github.com/freeplane/freeplane/discussions/401#discussioncomment-2845218). There, I explained that I partially solved this problem by forcing nodes to get minimized. But, my solution is very unorthodox.  The usual user, in this situation, does need to live with the long list of large children. 

>On the other hand, I love lists. My mind often will pour out a stream of more or less disjoint thoughts and I want to just get them down. In that case, actually and again just for me, I want to start at the top and go through them and refactor them some way, not keep them as a long list.

Yes, this kind of process works very well in outliners and mindmap software.  Some days ago, I was seeking for a creative solution for a problem, so I started with a simple node stating the problem, and, in just a few minutes, I had created more than a hundred nodes of pure brainstorming, rearranging and combining branches. This is very powerful.  So, yes, the grouping is a good practice, but when working with large nodes, the visibility gets complicated even with a few nodes, where grouping wouldn't make sense.

**Comment by awolo** [comment]
This is an interesting area you have opened up: as the number of nodes gets larger, you always have a tension between long lists - siblings, and long lineage - parents. That's where creativeity comes into play given the FP tools we have to work with. 

From an FP design point of view, the easiier it is to smoothly transform the map i either direction, the better. Does that capture your point? Or are you trying to formulate an FP design modification that will make this easier?

**Reply by euu2021** [reply]
>From an FP design point of view, the easiier it is to smoothly transform the map i either direction, the better. Does that capture your point?

Do you mean, my point in the first post? My point there is that having the parent in the middle increases the amount of nodes that can be shown on the screen at the same time as the parent. And this is good because the user can more easily see the context where the node is inserted.

**Comment by awolo** [comment]
Sorry, ther's a little typo in my prior answer.

Yes, but also, that sometimes we have more nodes, sometimes fewer. Sometimes lots of detail in the nodes, sometimes just a few words.

It's that we want FP to allow us to easily view the map in the way that gives us the most insight into what we want to see at the moment, which includes ancestors, siblings, contents of each node, and also relationships between them all.
